src/Symfony/Component/Workflow/Registry.php

```diff
@@ -28,8 +29,16 @@ class Registry
      */
     public function add(Workflow $workflow, $supportStrategy)
     {
-        if (!$supportStrategy instanceof SupportStrategyInterface) {
-            throw new \InvalidArgumentException('The "supportStrategy" is not an instance of SupportStrategyInterface.');
+        @trigger_error('add is deprecated since Symfony 4.0. Use addWorkflow instead.', E_USER_DEPRECATED);
+        $this->addWorkflow($workflow, $supportStrategy);
+    }
+
+    public function addWorkflow(WorkflowInterface $workflow, $supportStrategy)
+    {
+        if ($supportStrategy instanceof SupportStrategyInterface) {
+            @trigger_error('Passing a "SupportStrategyInterface" in "Registry::addWorkflow" is deprecated since 4.0. Use "SupportStrategyWorkflowInterface" instead.', E_USER_DEPRECATED);
+        } elseif (!($supportStrategy instanceof SupportStrategyWorkflowInterface || $supportStrategy instanceof SupportStrategyInterface)) {
```

Comment:
It looks like you did not simplified the condition.

Reply:
Yeah, sorry, that's modified.